var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -10,8 +10,8 @@ class Charliecloud(MakefilePackage):
     """Lightweight user-defined software stacks for HPC."""
 
     homepage = "https://hpc.github.io/charliecloud"
-    url      = "https://github.com/hpc/charliecloud/archive/v0.2.4.tar.gz"
```

Comment:
You should still keep a URL for the latest version here. It is needed for `spack versions`.

Reply:
@adamjstewart I have added this line back in my local branch. My thought here was that the assignment of url there was overriding `url_for_version` and causing the travis error. That was not the case.

I've read the Spack Packaging Guide's Version and Fetching section; it isn't clear to me how this function interacts with spack or why the unit tests fail now that it has been added. I'll keep tinkering.